ui/index.d.ts

```diff
@@ -1409,7 +1420,8 @@ export interface IMinimumServiceClientOptions {
     requestPolicyFactories?: any[] | ((defaultRequestPolicyFactories: any[]) => (void | any[]));
 }
 
-export type AzExtGenericClientInfo = AzExtServiceClientCredentials | { credentials: AzExtServiceClientCredentials; environment: Environment; } | undefined;
+export type AzExtGenericCredentials = BasicAuthenticationCredentials | TokenCredentials | AzExtServiceClientCredentials;
```

Comment:
Because we have two different createClient helper functions, `createGenericClient` and `createAzureClient`.

Originally the azureCredentials were similar enough to basic credential types, but since we're introducing `getToken` as a requirement to create azure clients to ensure T1&T2 compatability, it would force those requirements on any generic client we create 

For example, we used to make a GitHub client with `createGenericClient` but it wouldn't have required `getToken`.

I'm not sure if that 100% made sense so feel free to ping me and I can explain it better 😓